curriculum/challenges/english/20-upcoming-python/learn-list-comprehension-by-building-a-case-converter-program/657f43d341a0dd17120cdb08.md

```diff
@@ -0,0 +1,62 @@
+---
+id: 657f43d341a0dd17120cdb08
+title: Step 17
+challengeType: 20
+dashedName: step-17
+---
+
+# --description--
+
+Inside the space you left between the pair of square braces, you can describe your desired value based on a given condition.
+
+```py
+snake_cased_char_list = [
+    '_' + char.lower() if char.isupper()
+]
+```
+
+Python will interpret this expression as "append `'_' + char.lower()` to the list if `char` is in uppercase" and this covers the case for the capital letters in the input string.
+
+Start by adding this line within the square braces.
+
+# --hints--
+
+Test 1
+
+```js
+
+```
+
+# --seed--
+
+## --seed-contents--
+
+```py
+def convert_to_snake_case(pascal_or_camel_cased_string):
+    # snake_cased_char_list = []
+    # for char in pascal_or_camel_cased_string:
+    #     if char.isupper():
+    #         snake_cased_char_list.append(f'_{char.lower()}')
+    #     else:
+    #         snake_cased_char_list.append(char)
+    # snake_cased_string = ''.join(snake_cased_char_list)
+    # clean_snake_cased_string = snake_cased_string.strip('_')
+
+    # return clean_snake_cased_string
+
+    snake_cased_char_list = [
+--fcc-editable-region--
+
+
+
+--fcc-editable-region--
+    ]
+
+
+
```

Comment:
I would suggest removing one blank line to keep only two blank lines.

Reply:
Done